src/cascadia/QueryExtension/ILMProvider.idl

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILMProvider
+    {
+        // chat related functions
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+
+        // auth related functions
+        void SetAuthentication(Windows.Foundation.Collections.ValueSet authValues);
+        event Windows.Foundation.TypedEventHandler<ILMProvider, String> AuthChanged;
```

Comment:
`AzureOpenAILLMProvider` doesn't actually emit this event but the Github one is going to, so putting it in already